### PR TITLE
Feature/mava reproducibility and PZ wrapper fix

### PR DESCRIPTION
**Pull request by KaleabTessera**
## What?
- Added seeds to networks for better reproducibility.
- Fixed PZ wrapper issue. 
- Some cleanup and doc strings. 
## Why?
-
## How?
-
## Extra
-

### CODE REVIEW

File: mava/systems/tf/mad4pg/networks.py
Comment (DriesSmit) [inline mava/systems/tf/mad4pg/networks.py:70]: Did we merge MADDPG and MAD4PG's networks? It does seem like a great idea as much of the code is shared :+1: Thanks @KaleabTessera :smile:  It might be worth it to just check if both still work as expected.

File: mava/systems/tf/maddpg/networks.py
Comment (DriesSmit) [inline mava/systems/tf/maddpg/networks.py:166]: Should the + [1] be here in the case of MAD4PG? Probably not right?

Comment (DriesSmit) [reply mava/systems/tf/maddpg/networks.py:166]: Combining the networks for MADDPG and MAD4PG looks great thanks @KaleabTessera :raised_hands:

File: mava/components/tf/networks/continuous.py
Comment (DriesSmit) [inline mava/components/tf/networks/continuous.py:29]: What makes all these new components different from Acme? Is it to be able to use a seed?

**Review by DriesSmit — COMMENTED** [review]
Thanks, @KaleabTessera :fire: The changes look great. Just seem my few comments. I really like all the small code cleanups you did :smile:

File: mava/systems/tf/maddpg/networks.py
Comment (KaleabTessera) [reply mava/systems/tf/maddpg/networks.py:166]: Ah yes, that is a bug, thanks @DriesSmit !

File: mava/systems/tf/mad4pg/networks.py
Comment (KaleabTessera) [reply mava/systems/tf/mad4pg/networks.py:70]: I will run some benchmarks before merging this in :+1:

File: mava/components/tf/networks/continuous.py
Comment (KaleabTessera) [reply mava/components/tf/networks/continuous.py:29]: Yep. For most of this it is the seed and the networks are more configurable, i.e. layernorm and activations are configurable.

**Comment by DriesSmit** [comment]
Thanks

**Review by arnupretorius — APPROVED** [review]
Thanks @KaleabTessera! 💪  Looks great!

Happy to merge once we confirm the reproducibility of runs. 😄

File: mava/components/tf/networks/continuous.py
Comment (KaleabTessera) [reply mava/components/tf/networks/continuous.py]: Lol, fixed.

**Review by arnupretorius — APPROVED** [review]
Thanks for the tests and confirmation runs @KaleabTessera 🔥  This is great! 🥳

**Pull request merged**
